frontend/src/pages/discover.tsx

```diff
@@ -21,13 +23,65 @@ const fuseOptions = {
 const DiscoverPage: React.FC = () => {
   const [searchTerm, setSearchTerm] = useState("");
   const fuse = useMemo(() => new Fuse(psychiatrists, fuseOptions), []);
+
+  // Stores newSearchTerm in searchTerm
   const handleSearch = (newSearchTerm: string) => {
+    console.log(newSearchTerm);
     setSearchTerm(newSearchTerm);
   };
 
-  const searchResults = searchTerm
-    ? fuse.search(searchTerm).map(({ item }) => item)
-    : psychiatrists;
+  // Search function to search for first names, last names, and/or titles
+  const processSearch = () => {
+
+    const terms = searchTerm.trim().split(/\s+/);
+
+    // Handles searches with two terms
+    if (terms.length === 3) {
+      const [firstTerm, secondTerm, thirdTerm] = terms;
+      const results = psychiatrists.filter((psychiatrist) =>
+      ((psychiatrist.first_name.toLowerCase().includes(firstTerm.toLowerCase()) ||
+        psychiatrist.last_name.toLowerCase().includes(firstTerm.toLowerCase()) ||
+        psychiatrist.title.toLowerCase().includes(firstTerm.toLowerCase())) &&
+        (psychiatrist.first_name.toLowerCase().includes(secondTerm.toLowerCase()) ||
+          psychiatrist.last_name.toLowerCase().includes(secondTerm.toLowerCase()) ||
+          psychiatrist.title.toLowerCase().includes(secondTerm.toLowerCase())) &&
+        (psychiatrist.first_name.toLowerCase().includes(thirdTerm.toLowerCase()) ||
+          psychiatrist.last_name.toLowerCase().includes(thirdTerm.toLowerCase()) ||
+          psychiatrist.title.toLowerCase().includes(thirdTerm.toLowerCase())))
+      );
+      return results;
+    }
+    if (terms.length === 2) {
+      // Handles searches with two terms
+      const [firstTerm, secondTerm] = terms;
+      const results = psychiatrists.filter((psychiatrist) =>
+      ((psychiatrist.first_name.toLowerCase().includes(firstTerm.toLowerCase()) ||
+        psychiatrist.last_name.toLowerCase().includes(firstTerm.toLowerCase()) ||
+        psychiatrist.title.toLowerCase().includes(firstTerm.toLowerCase())) &&
+        (psychiatrist.first_name.toLowerCase().includes(secondTerm.toLowerCase()) ||
+          psychiatrist.last_name.toLowerCase().includes(secondTerm.toLowerCase()) ||
+          psychiatrist.title.toLowerCase().includes(secondTerm.toLowerCase())))
```

Comment:
you can use a higher order function for this, see the comment for an example above